### PR TITLE
Allow internal data function calls directly from our data class

### PR DESCRIPTION
**Pull request by JGSweets**
This allows the user to directly access functions of the wrapped data inside of our data class. e.g. if the data internally is a pandas dataframe:
```python
data = CSVData(...)  # has pandas dataframe
df_without_na= data.dropna() # will not be in place unless user specifies it

# or iterating through rows
for row in data.iterrows():
    print(row)
```

### CODE REVIEW

File: dataprofiler/tests/data_readers/test_base_data.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        # initialize the data class
+        data = BaseData(input_file_path="", data=FakeDataClass(), options={})
+
+        # if the function exists in BaseData fail the test because the results
+        # may become inaccurate.
+        self.assertFalse(hasattr(BaseData, 'func1'))
+
+        with self.assertRaisesRegex(AttributeError,
+                                    "'BaseData' nor 'FakeDataClass'"
+                                    "objects have no attribute 'test'"):
```
Comment (AnhTruong) [inline dataprofiler/tests/data_readers/test_base_data.py:32]: should be `"'Neither BaseData' nor 'FakeDataClass'"
                                    "objects has attribute 'test'"`

Comment (JGSweets) [reply dataprofiler/tests/data_readers/test_base_data.py:32]: fixed

File: dataprofiler/data_readers/base_data.py [hunk 2]
```diff
@@ -172,3 +172,33 @@ def length(self):
...
+        data.count(1)  # returns 2, bc data.data has the function 'count'
+        ```
+        """
+        try:
+            returned = object.__getattribute__(self, name)
+        except AttributeError as attr_error:
+            class_name = self.__class__.__name__
+            data_class_name = self.data.__class__.__name__
+            if (not f"'{class_name}' object has no attribute '{name}'"
+                    == str(attr_error)):
```
Comment (gme5078) [inline dataprofiler/data_readers/base_data.py:192]: Its seems strange to use error string for the conditional, but I'm not sure any other way to do this check

File: dataprofiler/tests/data_readers/test_base_data.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                                    "Neither 'BaseData' nor 'FakeDataClass' "
+                                    "objects have attribute 'test'"):
+            data.test
+
+        # validate it will take BaseData attribute over the data attribute
+        self.assertIsNone(data.data_type)
+
+        # validate will auto call the data function if it doesn't exist in
+        # BaseData
+        self.assertEqual("success", data.func1())
```
Comment (gme5078) [inline dataprofiler/tests/data_readers/test_base_data.py:40]: personally, did you try this with csv/pandas functions?

Comment (JGSweets) [reply dataprofiler/tests/data_readers/test_base_data.py:40]: yes.

File: dataprofiler/data_readers/base_data.py (hunk 2 above)
Comment (JGSweets) [reply dataprofiler/data_readers/base_data.py:192]: i validate it is an attribute error with the except and then validate the string, there might be a better way.

**Review by gme5078 — APPROVED** [review]

**Review by AnhTruong — APPROVED** [review]

**Pull request merged**